src/gnu/prolog/database/PrologTextLoader.java

```diff
@@ -328,7 +328,15 @@ protected void processModuleDirective(Term moduleName, Term exports)
 			//PrologException.typeError(TermConstants.atomAtom, moduleName);
 			return;
 		}
-		prologTextLoaderState.startNewModule(this, (AtomTerm)moduleName, exports);
+		try
+		{
+			prologTextLoaderState.startNewModule(this, (AtomTerm)moduleName, exports);
+		}
+		catch (PrologException e)
+		{
+			e.printStackTrace();
+			logError("unable to start new module");
```

Comment:
It could be so that you could 'catch' those errors in Java. I'm not really sure either - it was well before I had any involvement. I think that, generally speaking, fixing that is an orthogonal problem to implementing modules, and if you want to look at converting those logged messages into exceptions, we should file a new issue and do it in a new branch